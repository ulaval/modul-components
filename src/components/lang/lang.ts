import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
