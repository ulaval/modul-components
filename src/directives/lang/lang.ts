import Vue, { VNodeDirective } from 'vue';
import { PluginObject } from 'vue';
import { I18N_NAME } from '../directives-names';
import Messages from '@/utils/i18n';

export class MI18n extends Vue {
    public bind(element: HTMLElement, binding: VNodeDirective) {
        element.innerText = Messages.translate(binding.value);
    }
}

const MI18nPlugin: PluginObject<any> = {
    install(v, options) {
        v.directive(I18N_NAME, new MI18n());
    }
};

export default MI18nPlugin;
