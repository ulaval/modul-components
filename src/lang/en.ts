import { PluginObject } from 'vue';
import { BundleMessagesMap, ENGLISH, Messages } from '../utils/i18n/i18n';

const EnglishPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug('EnglishPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.en.json'),
                require('../components/accordion-group/accordion-group.lang.en.json'),
                require('../components/list-item/list-item.lang.en.json'),
                require('../components/modal/modal.lang.en.json'),
                require('../components/chip/chip.lang.en.json'),
                require('../components/datefields/datefields.lang.en.json'),
                require('../components/datepicker/datepicker.lang.en.json'),
                require('../components/daterangepicker/daterangepicker.lang.en.json'),
                require('../components/dropdown/dropdown.lang.en.json'),
                require('../components/dropdown-group/dropdown-group.lang.en.json'),
                require('../components/overlay/overlay.lang.en.json'),
                require('../components/file-select/file-select.lang.en.json'),
                require('../components/file-upload/file-upload.lang.en.json'),
                require('../components/icon-file/icon-file.lang.en.json'),
                require('../components/inplace-edit/inplace-edit.lang.en.json'),
                require('../components/input-style/input-style.lang.en.json'),
                require('../components/link/link.lang.en.json'),
                require('../components/limit-text/limit-text.lang.en.json'),
                require('../components/login/login.lang.en.json'),
                require('../components/message/message.lang.en.json'),
                require('../components/menu/menu.lang.en.json'),
                require('../components/option/option.lang.en.json'),
                require('../components/dialog/dialog.lang.en.json'),
                require('../components/pagination/pagination.lang.en.json'),
                require('../components/phone-number/phone-number.lang.en.json'),
                require('../components/repeater/repeater.lang.en.json'),
                require('../components/rich-text-editor/rich-text-editor.lang.en.json'),
                require('../components/table/table.lang.en.json'),
                require('../components/toast/toast.lang.en.json'),
                require('../components/tree/tree.lang.en.json'),
                require('../components/scroll-top/scroll-top.lang.en.json'),
                require('../components/show-more/show-more.lang.en.json'),
                require('../components/sidebar/sidebar.lang.en.json'),
                require('../components/spinner/spinner.lang.en.json'),
                require('../components/switch/switch.lang.en.json'),
                require('../components/textfield/textfield.lang.en.json'),
                require('../components/timepicker/timepicker.lang.en.json'),
                require('../components/tooltip/tooltip.lang.en.json'),
                require('../components/validation-message/validation-message.lang.en.json'),
                require('../filters/filesize/filesize.lang.en.json'),
                require('../filters/date/date/date.lang.en.json'),
                require('../filters/date/period/period.lang.en.json'),
                require('../filters/date/time/time.lang.en.json'),
                require('../filters/date/date-time/date-time.lang.en.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(ENGLISH, msgs));
        } else {
            throw new Error(
                'EnglishPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default EnglishPlugin;
