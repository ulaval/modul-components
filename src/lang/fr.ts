import { PluginObject } from 'vue';

import { BundleMessagesMap, FRENCH, Messages } from '../utils/i18n/i18n';

const FrenchPlugin: PluginObject<any> = {
    install(v, options): void {
        console.debug('FrenchPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype as any).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.fr.json'),
                require('../components/accordion-group/accordion-group.lang.fr.json'),
                require('../components/list-item/list-item.lang.fr.json'),
                require('../components/dialog/dialog.lang.fr.json'),
                require('../components/datefields/datefields.lang.fr.json'),
                require('../components/datepicker/datepicker.lang.fr.json'),
                require('../components/dropdown/dropdown.lang.fr.json'),
                require('../components/dropdown-item/dropdown-item.lang.fr.json'),
                require('../components/dropdown-group/dropdown-group.lang.fr.json'),
                require('../components/edit-window/edit-window.lang.fr.json'),
                require('../components/file-select/file-select.lang.fr.json'),
                require('../components/file-upload/file-upload.lang.fr.json'),
                require('../components/input-style/input-style.lang.fr.json'),
                require('../components/link/link.lang.fr.json'),
                require('../components/limit-text/limit-text.lang.fr.json'),
                require('../components/message/message.lang.fr.json'),
                require('../components/menu/menu.lang.fr.json'),
                require('../components/modal/modal.lang.fr.json'),
                require('../components/phone-number/phone-number.lang.fr.json'),
                require('../components/scroll-top/scroll-top.lang.fr.json'),
                require('../components/sidebar/sidebar.lang.fr.json'),
                require('../components/spinner/spinner.lang.fr.json'),
                require('../components/switch/switch.lang.fr.json'),
                require('../components/textfield/textfield.lang.fr.json'),
                require('../components/timepicker/timepicker.lang.fr.json'),
                require('../components/tooltip/tooltip.lang.fr.json'),
                require('../components/validation-message/validation-message.lang.fr.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(FRENCH, msgs));
        } else {
            throw new Error(
                'FrenchPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default FrenchPlugin;
