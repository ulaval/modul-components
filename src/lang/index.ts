/**
 * List public declaration here.
 */
export * from './fr';
