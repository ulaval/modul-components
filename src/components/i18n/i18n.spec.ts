import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent } from '../../../tests/helpers/render';
import MI18nPlugin, { MI18n } from './i18n';

describe('MI18n', () => {
    let localVue: VueConstructor<Vue>;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(MI18nPlugin);
        addMessages(localVue, ['components/i18n/i18n.spec.lang.fr.json']);
    });

    it('should render correctly', () => {
        const i18n = mount(MI18n, {
            localVue: localVue,
            propsData: {
                k: 'm-i18n-spec:a'
            }
        });

        return expect(renderComponent(i18n.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with params', () => {
        const i18n = mount(MI18n, {
            localVue: localVue,
            propsData: {
                k: 'm-i18n-spec:with-params',
                params: ['a', 'b']
            }
        });

        return expect(renderComponent(i18n.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with html encoding', () => {
        const i18n = mount(MI18n, {
            localVue: localVue,
            propsData: {
                k: 'm-i18n-spec:html-encode',
                params: ['<i>', '</i>'],
                'html-encode': true
            }
        });

        return expect(renderComponent(i18n.vm)).resolves.toMatchSnapshot();
    });

    it('should warn if key not defined', () => {
        jest.spyOn(console, 'warn');

        const i18n = mount(MI18n, {
            localVue: localVue,
            propsData: {
                k: 'undefined:key'
            }
        });

        expect(console.warn).toHaveBeenCalledWith(
            'The key undefined:key does not exist. Current lang: en'
        );
    });
});
