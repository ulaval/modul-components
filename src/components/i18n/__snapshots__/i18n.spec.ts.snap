// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MI18n should render correctly 1`] = `<span class="m-i18n">test a</span>`;

exports[`MI18n should render correctly with html encoding 1`] = `<span class="m-i18n">&lt;i&gt;test&lt;/i&gt;</span>`;

exports[`MI18n should render correctly with params 1`] = `<span class="m-i18n">test a, test b</span>`;
